### ISSUE TITLE
On OSX, Wine crashes when dealing with a lot a data

### ISSUE THREAD

**Issue by ineghi** [electron-userland/electron-builder#850]
- **Version**:  1.4.4

- **Target**: Windows

When building for Windows from OSX, I get this wine error;

`
Error: Exit code: 5. Command failed: wine /Users/ineghi/.cache/Squirrel.Windows/Squirrel.Windows-1.4.4/Writ
eZipToSetup.exe /Dev/Repos/mixer/dist/win/mixer Setup 0.0.0.exe /var/folders/db/nnpx_zv11nlgqt42df9h76p
r0000gn/T/electron-builder-be7-157fb17b72a-0/t-be7-2-setup.zip                                                    
wine: Unhandled page fault on read access to 0x00000000 at address 0x7b850832 (thread 0009), starting debugger... 
`

The application contains a lot of video (for a total of 4 go). This is probably a wine memory issue since everything is working fine without them.

edit: I use Wine 1.8.5 installed trough homebrew.

**Comment by develar** [comment]
You can use nsis target. "Consider to use nsis target for Windows (auto-update will be implemented this month) for new projects."

Set `build.win.target` to `nsis`. https://github.com/develar/onshape-desktop-shell/blob/nsis/package.json#L29

**Comment by ineghi** [comment]
Then, I'm running into this issue: [#503 ](https://github.com/electron-userland/electron-builder/issues/503)

**Comment by develar** [comment]
@ineghi Please attach actual error stack trace.

**Comment by ineghi** [comment]
Sorry :) there he is
`
Command line defined: "PROJECT_DIR=/Devarior/Repos/mixerpot"                                                      
Command line defined: "BUILD_RESOURCES_DIR=/Devarior/Repos/mixerpot/build"                                        
Command line defined: "APP_64=/Devarior/Repos/mixerpot/dist/mixerpot-0.0.0-x64.nsis.7z"                           
Command line defined: "RUN_AFTER_FINISH"                                                                          
Command line defined: "COMPRESS=auto"                                                                             
Command line defined: "ONE_CLICK"                                                                                 
Command line defined: "BUILD_UNINSTALLER"                                                                         
Command line defined: "UNINSTALLER_OUT_FILE=Z:\var\folders\db\nnpx_zv11nlgqt42df9h76pr0000gn\T\electron-builder-1e
cd-157fb4f0262-1\t-1ecd-1-uninstaller.exe"                                                                        
Processing script file: "<stdin>" (UTF8)                                                                          
                                                                                                                  
Error output:                                                                                                     
libc++abi.dylib: terminating with uncaught exception of type std::bad_alloc: std::bad_alloc                       
                                                                                                                  
    at ChildProcess.<anonymous> (/Devarior/Repos/mixerpot/node_modules/electron-builder/src/util/util.ts:160:14)  
    at ChildProcess.g (events.js:273:16)                                                                          
    at emitTwo (events.js:100:13)                                                                                 
    at ChildProcess.emit (events.js:185:7)                                                                        
    at maybeClose (internal/child_process.js:827:16)                                                              
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)                                      
From previous event:                                                                                              
    at NsisTarget.<anonymous> (/Devarior/Repos/mixerpot/node_modules/electron-builder/src/targets/nsis.ts:285:5)  
From previous event:                                                                                              
    at tsAwaiter (/Devarior/Repos/mixerpot/node_modules/electron-builder/src/util/awaiter.ts:10:47)               
    at Object.build (/Devarior/Repos/mixerpot/node_modules/electron-builder/out/builder.js:138:12)                
    at Object.<anonymous> (/Devarior/Repos/mixerpot/node_modules/electron-builder/out/build-cli.js:23:11)         
    at Module._compile (module.js:413:34)                                                                         
    at Object.Module._extensions..js (module.js:422:10)                                                           
    at Module.load (module.js:357:32)                                                                             
    at Function.Module._load (module.js:314:12)                                                                   
    at Function.Module.runMain (module.js:447:10)                                                                 
    at startup (node.js:141:18)                                                                                   
    at node.js:933:3
`

**Comment by develar** [comment]
Nice. Are you sure that you want to provide so big installer? Maybe will be better to download these huge video files during install (NSIS can this)?

**Comment by ineghi** [comment]
If you can download them from a local folder, that could be a viable solution. 
Sadly, the issue with nsis is not related to the size of the installer, since it's still failing after removing all the space consuming datas

**Comment by ineghi** [comment]
I solved the problem related to NSIS. Was caused by disk permissions errors... 
Anyway, It was without the datas. I can at least try to locally download these datas with the NSISdl plugin (I guess).

When I try the build with NSIS with the datas, I get this(after like 1 hour, literally)

```
Error: /Users/antoineghigny/.cache/nsis/nsis-3.0.1/mac/makensis exited with code 1
Output:
Command line defined: "APP_ID=com.electron.mixerpot"
Command line defined: "APP_GUID=6737475d-c56d-55b3-b44c-bd2b22cbe201"
Command line defined: "PRODUCT_NAME=mixerpot"
Command line defined: "PRODUCT_FILENAME=mixerpot"
Command line defined: "APP_DESCRIPTION=Try to implement electron builder"
Command line defined: "VERSION=0.0.0"
Command line defined: "COMPANY_NAME=Devarior"
Command line defined: "PROJECT_DIR=/Devarior/Repos/mixerpot"
Command line defined: "BUILD_RESOURCES_DIR=/Devarior/Repos/mixerpot/build"
Command line defined: "APP_64=/Devarior/Repos/mixerpot/dist/mixerpot-0.0.0-x64.nsis.7z"
Command line defined: "RUN_AFTER_FINISH"
Command line defined: "COMPRESS=auto"
Command line defined: "ONE_CLICK"
Command line defined: "UNINSTALLER_OUT_FILE=/var/folders/db/nnpx_zv11nlgqt42df9h76pr0000gn/T/electron-builder-be91-157fddfb324-0/t-be91-1-uninstaller.exe"
Processing script file: "<stdin>" (UTF8)
File: failed creating mmap of "/Devarior/Repos/mixerpot/dist/mixerpot-0.0.0-x64.nsis.7z"
!include: error in script: "installSection.nsh" on line 84
Error in script "<stdin>" on line 59 -- aborting creation process

    at ChildProcess.<anonymous> (/Devarior/Repos/mixerpot/node_modules/electron-builder/src/util/util.ts:160:14)
    at ChildProcess.g (events.js:273:16)
    at emitTwo (events.js:100:13)
    at ChildProcess.emit (events.js:185:7)
    at maybeClose (internal/child_process.js:827:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
From previous event:
    at NsisTarget.<anonymous> (/Devarior/Repos/mixerpot/node_modules/electron-builder/src/targets/nsis.ts:285:5)
From previous event:
    at tsAwaiter (/Devarior/Repos/mixerpot/node_modules/electron-builder/src/util/awaiter.ts:10:47)
    at Object.build (/Devarior/Repos/mixerpot/node_modules/electron-builder/out/builder.js:138:12)
    at Object.<anonymous> (/Devarior/Repos/mixerpot/node_modules/electron-builder/out/build-cli.js:23:11)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:141:18)
    at node.js:933:3
```

I continue digging :)

**Comment by develar** [comment]
I still think that it will better for your users if such video files will be downloaded during installation. e.g. you will upload it to Amazon S3. And installer will download it. Also, you will be able to release new version and do not duplicate such huge video files.

I can help you to implement "download during install", but I don't have time to implement feature "pack video files as is" Please see https://github.com/electron-userland/electron-builder/wiki/NSIS#custom-nsis-script — you can customize NSIS install script.

**Comment by ineghi** [comment]
Seems a good approach to me. But surprise, I tried again to build with NSIS and got (again) this error:
```
Error: /Users/dev/.cache/nsis/nsis-3.0.1/mac/makensis exited with code null                                                                                                                                                                
Output:                                                                                                                                                                                                                                              
Command line defined: "APP_ID=com.electron.mixer"                                                                                                                                                                                                 
Command line defined: "APP_GUID=6737475d-c56d-55b3-b44c-bd2b22cbe201"                                                                                                                                                                                
Command line defined: "PRODUCT_NAME=Mixer"                                                                                                                                                                                                        
Command line defined: "PRODUCT_FILENAME=Mixer"                                                                                                                                                                                                    
Command line defined: "APP_DESCRIPTION=Try to implement electron builder"                                                                                                                                                                            
Command line defined: "VERSION=0.0.0"                                                                                                                                                                                                                
Command line defined: "COMPANY_NAME=Dev"                                                                                                                                                                                                        
Command line defined: "PROJECT_DIR=/Dev/Repos/mixer"                                                                                                                                                                                         
Command line defined: "BUILD_RESOURCES_DIR=/Dev/Repos/mixer/build"                                                                                                                                                                           
Command line defined: "MUI_ICON=/Dev/Repos/mixer/build/icon.ico"                                                                                                                                                                             
Command line defined: "MUI_UNICON=/Dev/Repos/mixer/build/icon.ico"                                                                                                                                                                           
Command line defined: "APP_64=/Deva/Repos/mixer/dist/Mixer-0.0.0-x64.nsis.7z"                                                                                                                                                              
Command line defined: "RUN_AFTER_FINISH"                                                                                                                                                                                                             
Command line defined: "COMPRESS=auto"                                                                                                                                                                                                                
Command line defined: "ONE_CLICK"                                                                                                                                                                                                                    
Command line defined: "BUILD_UNINSTALLER"                                                                                                                                                                                                            
Command line defined: "UNINSTALLER_OUT_FILE=Z:\var\folders\db\nnpx_zv11nlgqt42df9h76pr0000gn\T\electron-builder-bfd-1581bd8d21c-0                                                                                                                    
\t-bfd-1-uninstaller.exe"                                                                                                                                                                                                                            
Processing script file: "<stdin>" (UTF8)                                                                                                                                                                                                             
                                                                                                                                                                                                                                                     
Error output:                                                                                                                                                                                                                                        
libc++abi.dylib: terminating with uncaught exception of type std::bad_alloc: std::bad_alloc                                                                                                                                                          
                                                                                                                                                                                                                                                     
    at ChildProcess.<anonymous> (/Dev/Repos/mixer/node_modules/electron-builder/src/util/util.ts:165:14)                                                                                                                                     
    at ChildProcess.g (events.js:273:16)                                                                                                                                                                                                             
    at emitTwo (events.js:100:13)                                                                                                                                                                                                                    
    at ChildProcess.emit (events.js:185:7)                                                                                                                                                                                                           
    at maybeClose (internal/child_process.js:827:16)                                                                                                                                                                                                 
    at Socket.<anonymous> (internal/child_process.js:319:11)                                                                                                                                                                                         
    at emitOne (events.js:90:13)                                                                                                                                                                                                                     
    at Socket.emit (events.js:182:7)                                                                                                                                                                                                                 
    at Pipe._onclose (net.js:475:12)                                                                                                                                                                                                                 
From previous event:                                                                                                                                                                                                                                 
    at /Dev/Repos/mixer/node_modules/electron-builder/src/targets/nsis.ts:282:5                                                                                                                                                              
From previous event:                                                                                                                                                                                                                                 
    at NsisTarget.executeMakensis (/Dev/Repos/mixer/node_modules/electron-builder/out/targets/nsis.js:324:11)                                                                                                                                
    at /Dev/Repos/mixer/node_modules/electron-builder/src/targets/nsis.ts:189:62                                                                                                                                                             
From previous event:                                                                                                                                                                                                                                 
    at NsisTarget.buildInstaller (/Dev/Repos/mixer/node_modules/electron-builder/out/targets/nsis.js:245:11)                                                                                                                                 
    at NsisTarget.finishBuild (/Dev/Repos/mixer/node_modules/electron-builder/src/targets/nsis.ts:66:49)                                                                                                                                     
    at /Dev/Repos/mixer/node_modules/electron-builder/src/packager.ts:138:31                                                                                                                                                                 
From previous event:                                                                                                                                                                                                                                 
    at Packager.doBuild (/Dev/Repos/mixer/node_modules/electron-builder/out/packager.js:254:11)                                                                                                                                              
    at /Dev/Repos/mixer/node_modules/electron-builder/src/packager.ts:102:32                                                                                                                                                                 
    at next (native)                                                                                                                                                                                                                                 
    at processImmediate [as _immediateCallback] (timers.js:383:17)                                                                                                                                                                                   
From previous event:                                                                                                                                                                                                                                 
    at Packager.build (/Dev/Repos/mixer/node_modules/electron-builder/out/packager.js:206:11)                                                                                                                                                
    at /Dev/Repos/mixer/node_modules/electron-builder/src/builder.ts:234:33                                                                                                                                                                  
    at next (native)                                                                                                                                                                                                                                 
From previous event:                                                                                                                                                                                                                                 
    at build (/Dev/Repos/mixer/node_modules/electron-builder/out/builder.js:81:21)                                                                                                                                                           
    at Object.<anonymous> (/Dev/Repos/mixer/node_modules/electron-builder/out/build-cli.js:66:41)                                                                                                                                            
    at Module._compile (module.js:413:34)                                                                                                                                                                                                            
    at Object.Module._extensions..js (module.js:422:10)                                                                                                                                                                                              
    at Module.load (module.js:357:32)                                                                                                                                                                                                                
    at Function.Module._load (module.js:314:12)                                                                                                                                                                                                      
    at Function.Module.runMain (module.js:447:10)                                                                                                                                                                                                    
    at startup (node.js:141:18)                                                                                                                                                                                                                      
    at node.js:933:3                  
```

I dig the internet and I think it's not related to electron builder, but I can't make it work tho...

**Comment by develar** [comment]
Previously you said that it "Was caused by disk permissions errors... ". Current error if you have a lot of video files or for plain small app?

**Comment by ineghi** [comment]
It makes this error with and without the files :-) 
I thought first it was related to disk permissions as the only time it worked was just after I run some disk reparations. It was apparently pure hazard since it's still failing now.

**Comment by Avien** [comment]
@ineghi I'm also experiencing the same problem
Where you able to solve it?
Thanks

**Comment by Avien** [comment]
@develar CanI get any advise please?
I am unable to build for NSIS on any platform I try
On ubuntu it simply hangs
On mac I get this error saying
```makensis exited with code null```

**Comment by ineghi** [comment]
@Avien Sorry, I've not managed to make it work. 
I came back to squirrel and manually injected the dependencies (And that's a poor solution ^^)

**Comment by rjb8682** [comment]
@Avien I'm also having this issue on a Mac Mini running Sierra 10.12.1.

**Comment by develar** [comment]
@rjb8682 Do you have enough disk space?

**Comment by rjb8682** [comment]
Oh, yes! I have about 1.03 TB free of 1.11TB

**Comment by develar** [comment]
@rjb8682 I cannot reproduce "libc++abi.dylib: terminating with uncaught exception of type std::bad_alloc: std::bad_alloc" — works on all my macs and CI macOS machines. It is not easy for me to help you. Maybe you can try to remove `~/Library/Caches/electron-builder`

**Comment by rjb8682** [comment]
@develar I completely understand! It seems like a weird issue to have. I deleted that cache and am re-running the build. I'll get back to you with what we get!

@McPo I'll give that a look as well!

**Comment by rjb8682** [comment]
Alright, still failed with the same error. I also looked at the `locale` command that was mentioned in #503 and it looks fine on my system. This does seem like a rather finicky issue, though

**Comment by McPo** [comment]
@rjb8682

Add the `script` entry to your build config as detailed here https://github.com/electron-userland/electron-builder/wiki/Options#buildnsis

Have it reference a file containing the following.

```
Unicode false

Section
SectionEnd
```

If that works (It wont generate an installer but it shouldn't throw an error), then change it to 
```
Unicode true

Section
SectionEnd
```

If it throws an error, then your issue is caused by unicode support, if not, thens its caused by some other issue.

**Comment by rjb8682** [comment]
@McPo @develar Fixed the issue! Looks like the locale thing was the problem. The CI environment was different than my local environment and the locale was not set in the CI environment. 

I appreciate all the help, guys!

**Comment by develar** [comment]
I will close issue as std::bad_alloc: std::bad_alloc should be fixed in next electron-builder release https://sourceforge.net/p/nsis/patches/271/

Feature: "download big files" using http://nsis.sourceforge.net/Inetc_plug-in plugin — feel free to post new issue.

**Comment by Avien** [comment]
I've upgraded to 10.8.1 and I still get the same error:
```Error: /Users/avi/Library/Caches/electron-builder/nsis/nsis-3.0.4/mac/makensis exited with code null
Output:
Command line defined: "APP_ID=ai.chorus.electron"
Command line defined: "APP_GUID=4c4314b6-a719-5458-bee0-d4e2843d6771"
Command line defined: "PRODUCT_NAME=Chorus"
Command line defined: "PRODUCT_FILENAME=Chorus"
Command line defined: "APP_FILENAME=Chorus"
Command line defined: "APP_DESCRIPTION=Note taking"
Command line defined: "VERSION=0.1.0"
Command line defined: "COMPANY_NAME=Avi Engelshtein"
Command line defined: "PROJECT_DIR=/Users/avi/Projects/AffectLayer/GitLab/desktop-app"
Command line defined: "BUILD_RESOURCES_DIR=/Users/avi/Projects/AffectLayer/GitLab/desktop-app/build"
Command line defined: "APP_32=/Users/avi/Projects/AffectLayer/GitLab/desktop-app/dist/Chorus-0.1.0-ia32.nsis.7z"
Command line defined: "INSTALL_MODE_PER_ALL_USERS"
Command line defined: "INSTALL_MODE_PER_ALL_USERS_REQUIRED"
Command line defined: "RUN_AFTER_FINISH"
Command line defined: "COMPRESS=auto"
Command line defined: "ONE_CLICK"
Command line defined: "BUILD_UNINSTALLER"
Command line defined: "UNINSTALLER_OUT_FILE=Z:\var\folders\w7\pywxqrln623b1x1_y0dx39nw0000gn\T\electron-builder-xsldxB\0-1-uninstaller.exe"
Processing script file: "<stdin>" (UTF8)```

**Comment by Rawnly** [comment]
Still got the same problem 

```
Error: /Users/rawnly/Library/Caches/electron-builder/nsis/nsis-3.0.4/mac/makensis exited with code null
Output:
Command line defined: "APP_ID=com.rawnly.splash"
Command line defined: "APP_GUID=89f94577-cc13-5f28-b4fd-a1e60cc41ba1"
Command line defined: "PRODUCT_NAME=splash"
Command line defined: "PRODUCT_FILENAME=splash"
Command line defined: "APP_FILENAME=splash"
Command line defined: "APP_DESCRIPTION=splash-cli GUI application"
Command line defined: "VERSION=1.0.0"
Command line defined: "COMPANY_NAME=Federico Vitale"
Command line defined: "PROJECT_DIR=/Volumes/Mini/Code/ELECTRON/splash"
Command line defined: "BUILD_RESOURCES_DIR=/Volumes/Mini/Code/ELECTRON/splash/build"
```

**Comment by maotora** [comment]
I'm getting the wine issue here using ArchLinux it's weird that the issue appears randomly, (sometimes it does sometimes I go through). But now it sticks and I can't build anymore.

```
> Mijengo@0.9.0 private:build:win /home/user/Projects/Electron/Mijengo/desktop
> build --win

electron-builder 19.35.1
No native production dependencies
Packaging for win32 x64 using electron 1.8.1 to dist/win-unpacked
Error: Exit code: null. Command failed: wine /home/user/.cache/electron-builder/winCodeSign/winCodeSign-1.9.0/rcedit.exe /home/user/Projects/Electron/Mijengo/desktop/dist/win-unpacked/Mijengo.exe --set-version-string FileDescription Mijengo --set-version-string ProductName Mijengo --set-version-string LegalCopyright Copyright © 2017 Hack EAC --set-file-version 0.9.0 --set-product-version 0.9.0.0 --set-version-string InternalName Mijengo --set-version-string OriginalFilename  --set-version-string CompanyName Hack EAC --set-icon /home/user/Projects/Electron/Mijengo/desktop/build/icon.ico
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 001a, blocked by 001d, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0019, blocked by 001d, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0028, blocked by 0027, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 001a, blocked by 001d, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0019, blocked by 001d, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0028, blocked by 0027, retrying (60 sec)

err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 001a, blocked by 001d, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0019, blocked by 001d, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0028, blocked by 0027, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 001a, blocked by 001d, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0019, blocked by 001d, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7bd06d80 "../../../wine/dlls/ntdll/loader.c: loader_section" wait timed out in thread 0028, blocked by 0027, retrying (60 sec)

    at /home/user/Projects/Electron/Mijengo/desktop/node_modules/builder-util/src/util.ts:116:16
    at ChildProcess.exithandler (child_process.js:277:5)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:927:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
From previous event:
    at exec (/home/user/Projects/Electron/Mijengo/desktop/node_modules/builder-util/src/util.ts:84:3)
    at wineExecutable.value.then.wine (/home/user/Projects/Electron/Mijengo/desktop/node_modules/builder-util/src/wine.ts:55:21)
From previous event:
    at execWine (/home/user/Projects/Electron/Mijengo/desktop/node_modules/builder-util/src/wine.ts:55:8)
    at /home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/src/winPackager.ts:334:11
From previous event:
    at WinPackager.signAndEditResources (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/out/winPackager.js:445:11)
    at WinPackager.signApp (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/src/winPackager.ts:350:17)
    at /home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/src/platformPackager.ts:208:16
From previous event:
    at WinPackager.doPack (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/out/platformPackager.js:261:11)
    at /home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/src/platformPackager.ts:114:16
    at Generator.next (<anonymous>)
From previous event:
    at WinPackager.pack (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/out/platformPackager.js:184:11)
    at /home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/src/packager.ts:260:24
From previous event:
    at Packager.doBuild (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/out/packager.js:344:11)
    at /home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/src/packager.ts:186:52
    at Generator.next (<anonymous>)
From previous event:
    at Packager.build (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/out/packager.js:263:11)
    at /home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/src/builder.ts:283:40
    at Generator.next (<anonymous>)
    at build (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/out/builder.js:63:21)
    at loadEnv.then (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/src/cli/cli.ts:49:4)
From previous event:
    at build (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/out/builder.js:63:21)
    at loadEnv.then (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/out/cli/cli.js:155:79)
    at runCallback (timers.js:781:20)
    at tryOnImmediate (timers.js:743:5)
    at processImmediate [as _immediateCallback] (timers.js:714:5)
From previous event:
    at Object.args [as handler] (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/out/cli/cli.js:155:68)
    at Object.runCommand (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/node_modules/yargs/lib/command.js:226:22)
    at Object.parseArgs [as _parseArgs] (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/node_modules/yargs/yargs.js:998:24)
    at Object.get [as argv] (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/node_modules/yargs/yargs.js:912:21)
    at Object.<anonymous> (/home/user/Projects/Electron/Mijengo/desktop/node_modules/electron-builder/out/cli/cli.js:151:418)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Function.Module.runMain (module.js:665:10)
    at startup (bootstrap_node.js:201:16)
    at bootstrap_node.js:626:3
npm ERR! code ELIFECYCLE
npm ERR! errno 255
npm ERR! Mijengo@0.9.0 private:build:win: `build --win`
npm ERR! Exit status 255
npm ERR!
npm ERR! Failed at the Mijengo@0.9.0 private:build:win script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2017-10-09T14_09_46_875Z-debug.log
ERROR: "private:build:win" exited with 255.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! Mijengo@0.9.0 pack:win: `run-s private:clean private:compile private:build:win`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the Mijengo@0.9.0 pack:win script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2017-10-09T14_09_46_966Z-debug.log
```

**Versions**

winetricks: 
`20170823 - sha256sum: 0663ccd4ee21dea5c4e3019dc52f9affd6d87dfb43d94bc2b87c0d1106d1356f`

wine: `wine-2.18`

Help!

**Comment by develar** [comment]
@maotora 

1. please try stable wine (2.0.2)
2. please try our proven and tested docker image to build: https://www.electron.build/multi-platform-build#docker
